packages/grid/x-data-grid-pro/src/hooks/features/rowPinning/useGridRowPinning.ts

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useGridApiMethod } from '@mui/x-data-grid';
+import { GridStateInitializer } from '@mui/x-data-grid/internals';
+
+import { GridApiPro } from '../../../models/gridApiPro';
+import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
+import { GridRowPinningApi } from './gridRowPinningInterface';
+
+export const rowPinningStateInitializer: GridStateInitializer<
+  Pick<DataGridProProcessedProps, 'pinnedRows' | 'getRowId'>
+> = (state, props, apiRef) => {
+  apiRef.current.unstable_caches.pinnedRows = {
+    top: [],
+    bottom: [],
+  };
+
+  return {
+    ...state,
+    rows: {
+      ...state.rows,
+      additionalRowGroups: {
+        ...state.rows?.additionalRowGroups,
+        pinnedRows: { top: [], bottom: [] },
+      },
+    },
+  };
+};
+
+export const useGridRowPinning = (
+  apiRef: React.MutableRefObject<GridApiPro>,
+  props: Pick<DataGridProProcessedProps, 'pinnedRows' | 'getRowId' | 'experimentalFeatures'>,
+): void => {
+  const isRowPinned = React.useCallback<GridRowPinningApi['unstable_isRowPinned']>(
+    (rowId) => {
+      return apiRef.current.getRowNode(rowId)?.isPinned || false;
+    },
+    [apiRef],
+  );
+
+  const setPinnedRows = React.useCallback<GridRowPinningApi['unstable_setPinnedRows']>(
+    (newPinnedRows) => {
+      if (!props.experimentalFeatures?.rowPinning) {
```

Comment:
How about introducing a `disableRowPinning` prop and force-setting it to `true` when the experimental feature is not enabled (like for aggregation)

But maybe it's not usefull since there is no way to set a pinned row through the UI

Reply:
The `disableRowPinning` would be here to stay after the feature is made stable
But due to the interconnection with the aggregation maybe it's risky to add a prop to fully disable it.
Or we should do it once we have a non-pinned version of the aggregation to fallback to.

Reply:
It would totally make sense when (if) we add a UI for row pinning, but I don't see the need for it right now